buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild/performance/tasks/PerformanceTest.groovy

```diff
@@ -117,38 +120,56 @@ abstract class PerformanceTest extends DistributionTest {
     }
 
     private boolean notContainsSpecialVersions() {
-        return ![configuredBaselines.getOrElse(""), determinedBaselines.getOrElse("")]
-            .collect { it.split(",") }
-            .flatten()
+        return !baselines.getOrElse("")
+            .split(",")
             .collect { it.toString().trim() }
             .any { NON_CACHEABLE_VERSIONS.contains(it) }
     }
 
     @Override
     @TaskAction
     void executeTests() {
+        if (getScenarios() == null) {
+            moveMethodFiltersToScenarios()
+        }
         try {
             super.executeTests()
         } finally {
             performanceReporter.report(this)
         }
     }
 
+    private void moveMethodFiltersToScenarios() {
+        DefaultTestFilter filter = getFilter() as DefaultTestFilter
+        List<String> scenarios = []
+        Set<String> onlyClassFilters = filter.getCommandLineIncludePatterns().collect { includePattern ->
+            def lastDot = includePattern.lastIndexOf(".")
+            if (lastDot == -1) {
+                return includePattern
+            } else {
+                def className = includePattern.substring(0, lastDot)
+                def methodName = includePattern.substring(lastDot + 1)
+                if (Character.isLowerCase(methodName.charAt(0))) {
+                    scenarios.add(methodName)
+                    return className
+                } else {
+                    return includePattern
+                }
+            }
+        } as Set
+        filter.setCommandLineIncludePatterns(onlyClassFilters)
+        setScenarios(scenarios.join(";"))
+    }
```

Comment:
This code is weird. We have a `collect()` call that also has side-effects. Then we mutate the inputs inside the task action. Can we straighten this out?

Reply:
I'll remove the side-effect from the `collect` call.

This is a workaround for test filters not working for unrolled spock tests. What I can do to make it work better is to remove the whole `scenarios` as an input, and only allow folks to use `--tests`. Then the above mutation of the inputs is `benign`, in the sense that it leaves the test filter in place and replaces it by something that works.

We have a fix for filtering unrolled Spock tests on `master`, though we didn't switch to the JUnit 5 platform, yet. As soon as we do that, the whole workaround can go away.

Reply:
I changed the collect call to `each` and added some documentation why we do this.